tests/GiraffeGenerator.IntegrationTests/SpecWithArgumentsTests.fs

```diff
@@ -87,23 +88,26 @@ type SpecWithArgumentsTests() =
     
     [<Fact>]
     let ``POST /abc/v1/records -> OK ["abc"]``() = task {
-        let! response = client.PostAsync("/abc/v1/records", null)
+        use content = new FormUrlEncodedContent(Seq.empty)
+        let! response = client.PostAsync("/abc/v1/records", content)
         let! text = response.Content.ReadAsStringAsync()
         Assert.Equal("[\"abc\"]",text)
         Assert.Equal(HttpStatusCode.OK, response.StatusCode)
     }
     
     [<Fact>]
     let ``POST /foo/v2/records -> OK ["good"]``() = task {
-        let! response = client.PostAsync("/foo/v2/records", null)
+        use content = new FormUrlEncodedContent(Seq.empty)
+        let! response = client.PostAsync("/foo/v2/records", content)
         let! text = response.Content.ReadAsStringAsync()
         Assert.Equal("[\"good\"]",text)
         Assert.Equal(HttpStatusCode.OK, response.StatusCode)
     }
     
     [<Fact>]
     let ``POST /abc/v3/records -> NOT_FOUND``() = task {
-        let! response = client.PostAsync("/abc/v3/records", null)
+        use content = new FormUrlEncodedContent(Seq.empty)
+        let! response = client.PostAsync("/abc/v3/records", content)
```

Comment:
btw, have we lost "non-form" tests?
I think we need both tests with form and without it

Reply:
Your original test (without empty form) would fail on the same spec without this change because endpoint declares that there should be a form and giraffe would fail the binding because of content-type mismatch.
There are some test cases for GET without any body content. Wouldn't they be enough? I believe this is not a big difference because HTTP verb changes only the routing generated and doesn't affect parameter bindings.

Reply:
Added path-only, query-only, route-only, form-only and json-only test cases